crates/ide/src/lib.rs

```diff
@@ -579,6 +586,33 @@ impl Analysis {
     }
 }
 
+fn add_ssr_assist(
+    db: &RootDatabase,
+    base: &mut Vec<Assist>,
+    resolve: bool,
+    frange: FileRange,
+) -> Option<()> {
+    let comment = {
+        let file = db.parse(frange.file_id);
+        file.tree().syntax().token_at_offset(frange.range.start()).find_map(ast::Comment::cast)
+    }?;
+    let match_finder = ide_ssr::ssr_from_comment(db, frange, &comment).ok()?;
+
+    let mut assist = Assist {
+        id: AssistId("ssr", AssistKind::RefactorRewrite),
+        label: Label::new("Apply SSR"),
```

Comment:
Any thoughts on having `Apply SSR in workspace` and `Apply SSR in file`? I think it could be nice to just apply it to a file while you are iterating, then once you've finalized the ssr rule apply it to the workspace.

Reply:
SGTM!

Ideally, we should have some kind of live preview for SSR, but that seems pretty far ahead.

Reply:
I've added the workspace/file options for this assist.